Userland/Libraries/LibWeb/HTML/SelectItem.h

```diff
@@ -8,31 +8,45 @@
 
 #include <AK/String.h>
 #include <LibIPC/Forward.h>
+#include <LibWeb/HTML/HTMLOptionElement.h>
 
 namespace Web::HTML {
 
-struct SelectItem {
-    enum class Type {
-        OptionGroup,
-        Option,
-        Separator,
-    };
-
-    Type type;
-    Optional<String> label = {};
-    Optional<String> value = {};
-    Optional<Vector<SelectItem>> items = {};
+struct SelectItemOption {
+    u32 id;
+    String label = {};
+    String value = {};
     bool selected = false;
+    bool disabled = false;
+    JS::Handle<HTMLOptionElement> option_element = {};
```

Comment:
Does this need to be a handle? Or can it just be a `GCPtr` / `NonnullGCPtr` that we visit in `visit_edges`?